### ISSUE TITLE
collisionStart firing multiple times

### ISSUE THREAD

**Issue by dxu** [liabru/matter-js#366]
Hi! I wrote up [a small plugin](https://github.com/dxu/matter-collisions) for adding collision events to bodies, but I'm looking to use this elsewhere, and I'm noticing that the 'collisionStart' event is being fired a lot of times compared what I would expect is a single collision between two boxes.

You can run the example included in the repo, but essentially I have two squares, `BoxA`  and `BoxB`, falling onto the ground at different heights. I expected that there would be 3 `collisionStart`'s (and 3 corresponding `collisionEnd`'s), while there might be a bunch of `collisionActive`'s:

1. `BoxA` hits Ground
2. `BoxB` hits `BoxA` (`BoxB` is above `BoxA`)
3. `BoxB` falls over, hits Ground.

But instead I get something along the lines of 10-20 collisionStarts and Ends.

![untitled](https://cloud.githubusercontent.com/assets/718423/22726736/e23ad26a-ed89-11e6-88e5-dfa665711332.png)

Is this intentional / expected? Am I doing something wrong?

Thanks for the great work you've done with this engine! It's really awesome, and I'm looking forward to playing with it further!

**Comment by dxu** [comment]
ah, never mind. I'm supposed to detect unique collisions via the pair ID, right?

**Comment by liabru** [comment]
I've not yet had chance to look into this - did you resolve it or was it a misunderstanding?

**Comment by dxu** [comment]
Oh, sorry! I thought I was just misunderstanding how the collisions were supposed to work. So if a box lands on its corner (assuming 0 restitution), and then falls flat onto the ground, `collisionStart` should only fire once in that case?

I'll reopen in case I'm misunderstanding my misunderstanding.

**Comment by liabru** [comment]
Yeah it should fire once, then multiple `collisionActive` then finally `collisionEnd`. Can you share your test case?

**Comment by dxu** [comment]
Here were the examples I was testing with, I console.log the events.

[Only `collisionStart`]( https://dxu.github.io/matter-test/start/)
[Only `collisionEnd`]( https://dxu.github.io/matter-test/end/)
[Only `collisionActive`](https://dxu.github.io/matter-test/active/)
[All events](https://dxu.github.io/matter-test/all/)

I get ~9-15 `collisionEnd` events (seems to be inconsistent across refreshes), ~12-17 `collisionStart` events

**Comment by liabru** [comment]
Thanks for this, will check it out.

**Comment by gokadi** [comment]
Hi! I faced the same issue. Are there any updates?  
    "matter-js": "^0.17.1",

**Comment by liabru** [comment]
@gokadi can you check `Matter.version` is it `0.18.0`?

If so can you try out the alpha build file included this PR? https://github.com/liabru/matter-js/pull/1079

Be great to know if it resolves, here's the discussion on it https://github.com/liabru/matter-js/issues/1077

**Comment by RandomGamingDev** [comment]
Hi! I faced the same issue. Are there any updates?
The same issue is happening to me on 0.19.0

**Comment by kekcsi** [comment]
Hi! I faced the same issue with 0.20.0.
